Document/0x05b-Basic-Security_Testing.md

```diff
@@ -207,6 +204,217 @@ Unlike static analysis, dynamic analysis is performed while executing the mobile
+
+```bash
+$ adb reverse tcp:8080 tcp:8080
+```
+
+Once you have done this all proxy traffic on your Android phone will be going to port 8080 on 127.0.0.1 and it will be redirected via adb to 127.0.0.1:8080 on your laptop and you will see now the traffic in your Burp. With this trick you are able to test and intercept traffic also in Wifis that have client isolation.
+
+
+##### Intercepting Non-Proxy Aware Apps
+
+Once you have setup an interception proxy and have a MITM position you might still not be able to see anything. This is mainly due to the following reasons:
+
+- The app is using a framework like Xamarin that simply is not using the proxy settings of the Android OS or
+- The app you are testing is verifying if a proxy is set and is not allowing now any communication.
+
+In both scenarios you would need additional steps to finally being able to see the traffic. In the sections below we are describing two possible solutions, ettercap and iptables. You could also use an access point that is under your control to redirect the traffic, but this would require additional hardware and we focus for now on software solutions.
+
+> For both solutions you need to activate "Support invisible proxying" in Burp, in Proxy Tab/Options/Edit Interface.
+
+**iptables**
+
+You can use iptables on the Android device to  redirect all traffic to your interception proxy. The following command would redirect port 80 to your proxy running on port 8080
+
+```bash
+$ iptables -t nat -A OUTPUT -p tcp --dport 80 -j DNAT --to-destination <Your-Proxy-IP>:8080
+```
+
+Verify the iptables settings and check the IP and port.
+
+```bash
+$ iptables -t nat -L
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+DNAT       tcp  --  anywhere             anywhere             tcp dpt:5288 to:<Your-Proxy-IP>:8080
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain natctrl_nat_POSTROUTING (0 references)
+target     prot opt source               destination
+
+Chain oem_nat_pre (0 references)
+target     prot opt source               destination
+```
+
+In case you want to reset the iptables configuration you can flush the rules:
+
+```bash
+$ iptables -t nat -F
+```
+
+**Ettercap**
+
+Read the chapter "Testing Network Communication" and the test case "Simulating a Man-in-the-Middle Attack" for further preparation and instructions for running ettercap.
+
+Your testing machine and the Android device must be connected to the same wireless network. Start ettercap with the following command, replacing the IP addresses below with the IP addresses of your Android device and the wireless network's gateway.
```

Comment:
Your testing machine --> The machine where you run your proxy

Reply:
done